blockchain/blocks/src/header.rs

```diff
@@ -294,14 +294,14 @@ impl BlockHeader {
         Ok(())
     }
     /// Check to ensure block signature is valid
-    pub fn check_block_signature(&self, addr: &Address) -> Result<(), Error> {
+    pub fn check_block_signature(&self, addr: Address) -> Result<(), Error> {
```

Comment:
Helped in an async context since dealing with references

Reply:
There is no need for it to be like this, I just checked